src/main.ts

```diff
@@ -277,13 +278,15 @@ export default class AttachmentManagementPlugin extends Plugin {
     const oldNotePath = path.dirname(oldPath);
     const oldNoteExtension = path.extname(oldPath);
     const oldNoteName = path.basename(oldPath, oldNoteExtension);
-
+    //generate parent of oldnotepath, last of the oldNotePath
+    const oldNoteParent = path.basename(path.dirname(oldPath));
+    
     debugLog("Old Note Path:", oldNotePath);
     debugLog("Old Note Name:", oldNoteName);
 
     // generate old attachment path
-    const oldAttachPath = this.getAttachmentPath(oldNoteName, oldNotePath, setting);
-    const newAttachPath = this.getAttachmentPath(rf.basename, rf.parent?.path as string, setting);
+    const oldAttachPath = this.getAttachmentPath(oldNoteName, oldNotePath, setting, oldNoteParent);
+    const newAttachPath = this.getAttachmentPath(rf.basename, rf.parent?.path as string, setting, rf.parent?.parent?.name as string);
```

Comment:
To prevent the use case that the parent was `null` and leading the parent.name was `undefined`, try to add this code
```ts
  const newNoteParentFolder = rf.parent;
  let newNotePath = "/";
  let newNoteParent = "/";
  if (newNoteParentFolder !== null) {
    newNotePath = newNoteParentFolder.path;
    newNoteParent = newNoteParentFolder.name;
  }
  const oldAttachPath = this.getAttachmentPath(oldNoteName, oldNotePath, oldNoteParent, setting);
  const newAttachPath = this.getAttachmentPath(rf.basename, newNotePath, newNoteParent, setting);
```